utils/oc.py

```diff
@@ -379,23 +390,25 @@ class OC_Map(object):
 
     def __init__(self, clusters=None, namespaces=None,
                  integration='', e2e_test='', settings=None,
-                 internal=None, use_jump_host=True):
+                 internal=None, use_jump_host=True, thread_pool_size=1,
+                 init_projects=False):
         self.oc_map = {}
         self.calling_integration = integration
         self.calling_e2e_test = e2e_test
         self.settings = settings
         self.internal = internal
         self.use_jump_host = use_jump_host
+        self.thread_pool_size = thread_pool_size
+        self.init_projects = init_projects
+        self._lock = Lock()
 
         if clusters and namespaces:
             raise KeyError('expected only one of clusters or namespaces.')
         elif clusters:
-            for cluster_info in clusters:
-                self.init_oc_client(cluster_info)
+            threaded.run(self.init_oc_client, clusters, self.thread_pool_size)
         elif namespaces:
-            for namespace_info in namespaces:
-                cluster_info = namespace_info['cluster']
-                self.init_oc_client(cluster_info)
+            clusters = [ns_info['cluster'] for ns_info in namespaces]
+            threaded.run(self.init_oc_client, clusters, self.thread_pool_size)
```

Comment:
the `threaded.run()` can be called after the `if` block to avoid duplication. That will remove the need for `elif clusters:`.

Reply:
i prefer to keep all the cases covered explicitly. wdyt?

Reply:
sure